code/modules/atmospherics/machinery/atmosmachinery.dm

```diff
@@ -91,12 +100,37 @@ Pipelines + Other Objects -> Pipe network
 			if(can_be_node(target, I))
 				NODE_I = target
 				break
-
 	update_icon()
 
+/obj/machinery/atmospherics/proc/setPipingLayer(new_layer)
+	piping_layer = new_layer
+	pixel_x = (piping_layer - PIPING_LAYER_DEFAULT) * PIPING_LAYER_P_X
+	pixel_y = (piping_layer - PIPING_LAYER_DEFAULT) * PIPING_LAYER_P_Y
+	layer = initial(layer) + ((piping_layer - PIPING_LAYER_DEFAULT) * PIPING_LAYER_LCHANGE)
+
 /obj/machinery/atmospherics/proc/can_be_node(obj/machinery/atmospherics/target)
-	if(target.initialize_directions & get_dir(target,src))
+	if(connection_check(target, piping_layer))
+		return TRUE
+	return FALSE

```

Comment:
seems redundant now, but i'm not sure how it's handled in the children. either remove this proc entirely and replace with `connection_check()` or change the contents of this proc to simply `return connection_check(target,piping_layer)` if for some reason both procs MUST exist.